lib/src/source/hosted.dart

```diff
@@ -946,80 +923,140 @@ class BoundHostedSource extends CachedSource {
   static const _prefetchingKey = #_prefetch;
 }
 
-/// This is the modified hosted source used when pub get or upgrade are run
-/// with "--offline".
-///
-/// This uses the system cache to get the list of available packages and does
-/// no network access.
-class _OfflineHostedSource extends BoundHostedSource {
-  _OfflineHostedSource(HostedSource source, SystemCache systemCache)
-      : super(source, systemCache);
+/// The [PackageName.description] for a [HostedSource], storing the package name
+/// and resolved URI of the package server.
+class HostedDescription extends Description<HostedDescription> {
+  final String packageName;
+  final String url;
```

Comment:
This is the URL to the package repository, right?
```suggestion
  /// Url for package repository containing [packageName] described in the this [Description].
  final String hostedUrl;
```

That way, we don't think this is the URL to package archive or something else :D